apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1067,6 +1070,28 @@ def _create_shadow_ext_net_for_nat(self, context, l3out_name, ext_epg_name,
                 self.apic_manager.ensure_external_epg_created(
                     shadow_l3out, external_epg=shadow_ext_epg,
                     owner=router_tenant, transaction=trs)
+
+                # if there is a router_type (like ASR) then we have to flesh
+                # out this shadow L3 out in APIC
+                if self._is_asr_router_type(net_info):
+                    address = net_info['cidr_exposed']
```

Comment:
Much of this code is repeated in create_real_external_network. Will it be worthwhile to factor it out?